service/history/workflow/update/update.go

```diff
@@ -143,6 +144,7 @@ func newAdmitted(id string, request *anypb.Any, opts ...updateOpt) *Update {
 		instrumentation: &noopInstrumentation,
 		accepted:        future.NewFuture[*failurepb.Failure](),
 		outcome:         future.NewFuture[*updatepb.Outcome](),
+		admittedTime:    time.Now(),
```

Comment:
What's the benefit of tracking "admitted time" instead of "requested time"? I believe the transition happens - if it happens - immediately and within the same workflow lock. It would eliminate the need for the code inside `Admit`, wouldn't it?

Reply:
`newAdmited` is only used to recreated update in `Admitted` state (i.e. reset). It is not used with normal flow.

Reply:
That's fair; I just attached my point to a code change.

Point I'm trying to make is: why does it have to be "admitted"?

Reply:
It doesn't have to be "admitted". But I think we'd have to add the "requested" time in the `New` and `newAdmitted` constructors so I'm not sure it saves that much code? Also "requested" isn't a prominent concept in update and could possibly be taken to mean the time of the client-side request. We can definitely change the implementation here, but I think I'll merge as-is seeing as we don't have correctness concerns.